Fixes #11775: Allow database ID for adding puppet modules to CV

ehelms opened the pull request.

jlsherrill commented on app/controllers/katello/api/v2/content_view_puppet_modules_controller.rb:
```diff
@@ -20,6 +20,7 @@ def index
     param :name, String, :desc => N_("name of the puppet module")
     param :author, String, :desc => N_("author of the puppet module")
+    param :id, String, :desc => N_("the id of the puppet module to associate")
     param :uuid, String, :desc => N_("the uuid of the puppet module to associate")
```
the find_puppet_module method only looks like it considers the id param.  Should it also consider the uuid param?

ehelms replied:
Thinking more about it, I don't know that this option would make sense so I am inclined to remove it (the 'id') option that I added since content view puppet modules are not connected to puppet modules at the database level.

ehelms replied:
The find_puppet_module function isn't for this create action so wouldn't need updating.

ehelms commented:
Updated to handle ID being passed in

jlsherrill commented on app/models/katello/content_view_puppet_module.rb:
```diff
@@ -49,7 +49,8 @@ def set_attributes
       elsif (self.name.present? && !self.author.present?)
         puppet_modules = PuppetModule.latest_modules_search(
```
does this function even exist?

ehelms replied:
Good catch

ehelms commented:
I updated to remove that function you commented on -- in reality it was an unreachable piece of code due to validators. I also added some tests to ensure creation of content view puppet modules occurred as we would expect.

jlsherrill commented:
Discovered and filed:  http://projects.theforeman.org/issues/11877

but unrelated to this PR.  so ACK
